javasource/oql/implementation/MxCSVWriter.java

```diff
@@ -0,0 +1,77 @@
+package oql.implementation;
+
+//Based on RFC 4180 and added support for an escape character if quotes are not used
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class MxCSVWriter implements Closeable {
+
+  protected Writer writer;
+  protected List<Character> specialCharactersToEscape;
+  protected char separatorChar;
+  protected Optional<Character> quoteChar;
+  protected Optional<Character> escapeChar;
+  protected String lineEnd = "\r\n";
+
+  public MxCSVWriter(Writer writer, char separatorChar, Optional<Character> quoteChar, Optional<Character> escapeChar) {
+    this.writer = writer;
+    this.separatorChar = separatorChar;
+    this.quoteChar = quoteChar;
+    this.escapeChar = escapeChar;
+    specialCharactersToEscape = defineSpecialCharactersToEscape();
+  }
+
+  public void writeRow(String[] columns) throws IOException {

```

Comment:
I think we should switch from arrays to lists here: both `IDataTable.getRows()` and `IDataTableSchema.getColumnSchemas()` are lists, so they would be easier to convert